internal/cli/cmd/dac/diff/diff.go

```diff
@@ -82,34 +98,61 @@ func (o *option) Validate() error {
 }
 
 func (o *option) Execute() error {
-	for _, preview := range o.dashboards {
-		project := resource.GetProject(preview.GetMetadata(), o.Project)
-		dashboard, err := o.apiClient.V1().Dashboard(project).Get(preview.Metadata.Name)
-		if err != nil {
-			return err
-		}
+	// Create the output folder (+ any parent folder if applicable) where to store the diff files
+	err := os.MkdirAll(config.Global.Dac.OutputFolder, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("error creating the output folder: %v", err)
+	}
 
-		d, err := dashboardDiff(dashboard, preview)
-		if err != nil {
-			return err
-		}
+	var result []diffResult
+	for _, updatedDashboard := range o.dashboards {
+		project := resource.GetProject(updatedDashboard.GetMetadata(), o.Project)
+		status, filePath := o.processDashboardDiff(updatedDashboard, project)
 
-		// Create the folder (+ any parent folder if applicable) where to store the output
-		err = os.MkdirAll(config.Global.Dac.OutputFolder, os.ModePerm)
-		if err != nil {
-			return fmt.Errorf("error creating the output folder: %v", err)
-		}
+		result = append(result, diffResult{
+			Dashboard: updatedDashboard.Metadata.Name,
+			Project:   project,
+			Status:    status,
+			Diff:      filePath,
+		})
 
-		filePath := path.Join(config.Global.Dac.OutputFolder, fmt.Sprintf("%s-%s.diff", project, dashboard.Metadata.Name))
-		if writeErr := os.WriteFile(filePath, []byte(d), 0644); writeErr != nil { // nolint: gosec
-			return fmt.Errorf("unable to write the diff file: %w", writeErr)
+		if filePath != "" {
+			logrus.Infof("%s successfully generated", filePath)
 		}
+	}
 
-		if outputErr := output.HandleString(o.writer, fmt.Sprintf("%s successfully created", filePath)); outputErr != nil {
-			return outputErr
+	return output.Handle(o.writer, o.Output, result)
+}
+
+func (o *option) processDashboardDiff(updatedDashboard *modelV1.Dashboard, project string) (string, string) {
+	currentDashboard, err := o.apiClient.V1().Dashboard(project).Get(updatedDashboard.Metadata.Name)
+	if err != nil {
+		var reqErr *perseshttp.RequestError
+		if errors.As(err, &reqErr) {
```

Comment:
Instead of casting the err, you could simply do :

```suggestion
 if errors.Is(createErr, perseshttp.RequestNotFoundError) {
}
```